mono/metadata/threadpool-worker-default.c

```diff
@@ -123,10 +123,10 @@ typedef union {
 typedef struct {
 	MonoRefCount ref;
 
-	MonoThreadPoolWorkerCallback callback;
-
 	ThreadPoolWorkerCounter counters;
 
+	MonoThreadPoolWorkerCallback callback;
+
```

Comment:
I think you'll also need to ensure the ThreadPoolWorker struct itself is aligned to 64-bits?